lib/template.js

```diff
@@ -0,0 +1,108 @@
+const jq = require('./jq');
+
+const findInsideDoubleBracesIndices = (input) => {
+  let wrappingQuote = null;
+  let insideDoubleBracesStart = null;
+  const indices = [];
+
+  for (let i = 0; i < input.length; i += 1) {
+    const char = input[i];
+
+    if (char === '"' || char === "'") {
+      // If inside quotes, ignore braces
+      if (!wrappingQuote) {
+        wrappingQuote = char;
+      } else if (wrappingQuote === char) {
+        wrappingQuote = null;
+      }
+    } else if (!wrappingQuote && char === '{' && i > 0 && input[i - 1] === '{') {
+      // if opening double braces that not wrapped with quotes
+      if (insideDoubleBracesStart) {
+        throw new Error(`Found double braces in index ${i - 1} inside other one in index ${insideDoubleBracesStart - '{{'.length}`);
+      }
+      insideDoubleBracesStart = i + 1;
+      if (input[i + 1] === '{') {
+        // To overcome three "{" in a row considered as two different opening double braces
+        i += 1;
+      }
+    } else if (!wrappingQuote && char === '}' && i > 0 && input[i - 1] === '}') {
+      // if closing double braces that not wrapped with quotes
+      if (insideDoubleBracesStart) {
+        indices.push({start: insideDoubleBracesStart, end: i - 1});
+        insideDoubleBracesStart = null;
+        if (input[i + 1] === '}') {
+          // To overcome three "}" in a row considered as two different closing double braces
+          i += 1;
+        }
+      } else {
+        throw new Error(`Found closing double braces in index ${i - 1} without opening double braces`);
+      }
+    }
+  }
+
+  if (insideDoubleBracesStart) {
+    throw new Error(`Found opening double braces in index ${insideDoubleBracesStart - '{{'.length} without closing double braces`);
+  }
+
+  return indices;
+}
+
+const render = (inputJson, template) => {
+  if (typeof template !== 'string') {
+    return null;
+  }
+  const indices = findInsideDoubleBracesIndices(template);
+  if (!indices.length) {
+    // If no jq templates in string, return it
+    return template;
+  }
+
+  const firstIndex = indices[0];
+  if (template.trim().startsWith('{{') && template.trim().endsWith('}}')) {
```

Comment:
is it required?
I believe the foreach will solve that as well, doesn't it?

Reply:
Yes it's required because if all the string is template we keep the jq result type. Otherwise we do json stringify. Added a comment to explain it